### PR TITLE
Make repository analysis API available to non-operators

### LINKED ISSUE #100318: Repository analysis API should be available to non-operators
When it was first released, we made the [repository analysis API](https://www.elastic.co/guide/en/elasticsearch/reference/current/repo-analysis-api.html) an operator-only API because we did not think it made sense for users (including cluster admins) to run it in ESS, and we expected it to be relatively normal for an ECE admin to have operator-level access. This reasoning doesn't really hold true in practice: even on ESS folks are connecting random third-party repositories sometimes, and ECE admins don't have operator-level access as a rule. I think we should remove the operator-only restriction.

I do not expect this to have any billing implications: the analysis API uses the same underlying repository as everything else so the requests it makes will be metered and billed as normal. It might have some security implications, we only really want this API to be accessible to high-privileged users as it can generate traffic and run up bills quite quickly.

Unfortunately the 60s proxy timeout limits the usefulness of this API in ECE and ESS, because 60s is not enough for a thorough analysis run. But it is at least a start.

### PR DESCRIPTION
**Pull request by nicktindall**
Closes #100318

### CODE REVIEW

**Comment by ywangd** [comment]
It seems to me that the existing tests for this API do _not_ test whether it is operator-only. As far as I can tell, all test clusters have security disabled. I think ideally we should have one of them enable security and configure a non-operator user to show the change is effective.

**Comment by nicktindall** [comment]
> It seems to me that the existing tests for this API do _not_ test whether it is operator-only. As far as I can tell, all test clusters have security disabled. I think ideally we should have one of them enable security and configure a non-operator user to show the change is effective.

There is a test which goes through all the actions and verifies that they are either operator-only or explicitly specified as no operator only. (`org.elasticsearch.xpack.security.operator.OperatorPrivilegesIT#testEveryActionIsEitherOperatorOnlyOrNonOperator`)

But as you point out there is no integration test which exercises the expected behaviour.

File: docs/reference/snapshot-restore/apis/repo-analysis-api.asciidoc [hunk 1]
```diff
@@ -39,9 +39,6 @@ POST /_snapshot/my_repository/_analyze?blob_count=10&max_blob_size=1mb&timeout=1
 <<privileges-list-cluster,cluster privilege>> to use this API. For more
 information, see <<security-privileges>>.
 
-* If the <<operator-privileges,{operator-feature}>> is enabled, only operator
-users can use this API.
-
```
Comment (ywangd) [inline docs/reference/snapshot-restore/apis/repo-analysis-api.asciidoc:44]: It needs to be deleted from this page as well
https://github.com/elastic/elasticsearch/blob/2d45e47bb5abf66ad80a06a7abf60de26ce5f147/docs/reference/security/operator-privileges/operator-only-functionality.asciidoc?plain=1#L24

**Review by ywangd — APPROVED** [review]
LGTM

**Pull request merged**